### ISSUE TITLE
TypeError: Cannot read properties of undefined (reading 'filter')

### ISSUE THREAD

**Issue by MPV** [argocd-diff-action/argocd-diff-action#76]
Not sure yet if this is a duplicate of:
- #4 

...but raising a separate issue nonetheless to start with (in case it turns out to be another cause). 

---

I'm getting this error:
```javascript
TypeError: Cannot read properties of undefined (reading 'filter') 
Error: TypeError: Cannot read properties of undefined (reading 'filter')
     at AppCollection.filterByRepo (/home/runner/_work/_actions/argocd-diff-action/argocd-diff-action/25ca470eda1d9eb4fa38ca8c666ce3bf5a9098f4/dist/index.js:13471:44)
     at /home/runner/_work/_actions/argocd-diff-action/argocd-diff-action/25ca470eda1d9eb4fa38ca8c666ce3bf5a9098f4/dist/index.js:13875:14
     at Generator.next (<anonymous>)
     at fulfilled (/home/runner/_work/_actions/argocd-diff-action/argocd-diff-action/25ca470eda1d9eb4fa38ca8c666ce3bf5a9098f4/dist/index.js:13771:58)
     at runMicrotasks (<anonymous>)
     at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

When running it like this:

```yaml
  argocd-app-diff:
    name: ArgoCD app diff
    steps:
      - name: Checkout ${{ github.repository }}
        uses: actions/checkout@a5ac7e51b41094c92402da3b24376905380afc29 # v4.1.6

      - uses: argocd-diff-action/argocd-diff-action@25ca470eda1d9eb4fa38ca8c666ce3bf5a9098f4 # 0.1.13
        env:
          KUBECTL_EXTERNAL_DIFF: diff -u
        with:
          argocd-server-fqdn: ${{ secrets.ARGOCD_URL }}
          argocd-token: ${{ secrets.ARGOCD_AUTH_TOKEN }}
          github-token: ${{ secrets.GITHUB_TOKEN }}
          argocd-extra-cli-args: --grpc-web
```

**Comment by MPV** [comment]
Or maybe it's rather a duplicate of: 🤔 
- https://github.com/argocd-diff-action/argocd-diff-action/issues/29

**Comment by MPV** [comment]
Seems like this change may solve this problem:
- https://github.com/argocd-diff-action/argocd-diff-action/pull/30

**Comment by MPV** [comment]
For what it's worth, I resolved my problem by making sure my repository could access the secret sent to the action using:
`argocd-token: ${{ secrets.ARGOCD_AUTH_TOKEN }}`

**Comment by Voziv** [comment]
🤔 , I wonder if this was a symptom of poor handling when the auth token was incorrect. I'm going to close this since we merged #30. If it's still an issue please feel free to re-open
